pkg/provider/onboardbase/provider.go

```diff
@@ -0,0 +1,143 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implieclient.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package onboardbase
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+
+	kclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	esv1beta1 "github.com/external-secrets/external-secrets/apis/externalsecrets/v1beta1"
+	dClient "github.com/external-secrets/external-secrets/pkg/provider/onboardbase/client"
+	"github.com/external-secrets/external-secrets/pkg/utils"
+)
+
+const (
+	errNewClient    = "unable to create OnboardbaseClient : %s"
+	errInvalidStore = "invalid store: %s"
+	errOnboardbaseStore = "missing or invalid Onboardbase SecretStore"
+)
+
+// Provider is a Onboardbase secrets provider implementing NewClient and ValidateStore for the esv1beta1.Provider interface.
+type Provider struct{}
+
+// https://github.com/external-secrets/external-secrets/issues/644
+var _ esv1beta1.SecretsClient = &Client{}
+var _ esv1beta1.Provider = &Provider{}
+
+func init() {
+	esv1beta1.Register(&Provider{}, &esv1beta1.SecretStoreProvider{
+		Onboardbase: &esv1beta1.OnboardbaseProvider{},
+	})
+}
+
+func (p *Provider) Capabilities() esv1beta1.SecretStoreCapabilities {
+	return esv1beta1.SecretStoreReadOnly
+}
+
+func (p *Provider) NewClient(ctx context.Context, store esv1beta1.GenericStore, kube kclient.Client, namespace string) (esv1beta1.SecretsClient, error) {
+	storeSpec := store.GetSpec()
+
+	if storeSpec == nil || storeSpec.Provider == nil || storeSpec.Provider.Onboardbase == nil {
+		return nil, fmt.Errorf(errOnboardbaseStore)
+	}
+
+	onboardbaseStoreSpec := storeSpec.Provider.Onboardbase
+
+	// Default Key to onboardbaseToken if not specified
+	if onboardbaseStoreSpec.Auth.OnboardbaseAPIKey.Key == "" {
+		storeSpec.Provider.Onboardbase.Auth.OnboardbaseAPIKey.Key = "onboardbaseToken"
+	}
+
+	// Default Key to onboardbaseToken if not specified
+	if onboardbaseStoreSpec.Auth.OnboardbasePasscode.Key == "" {
+		storeSpec.Provider.Onboardbase.Auth.OnboardbasePasscode.Key = "onboardbasePasscode"
+	}
+
+
+	// Default Key to onboardbaseToken if not specified
+	if onboardbaseStoreSpec.Environment == "" {
+		storeSpec.Provider.Onboardbase.Environment = "development"
+	}
+
+
+	// Default Key to onboardbaseProject if not specified
+	if onboardbaseStoreSpec.Project == "" {
+		storeSpec.Provider.Onboardbase.Project = "test"
+	}
```

Comment:
you can set defaults through the CRD spec, [see here](https://github.com/external-secrets/external-secrets/blob/c5414567406279cfb06bda4e111a01c251861af5/apis/externalsecrets/v1beta1/secretstore_vault_types.go#L48) for example.